### PR TITLE
Lint with pre-commit and fix zizmor error

### PR DESCRIPTION
**Pull request by hugovk**
## Summary

Run zizmor on this repo using pre-commit on GitHub Actions, and add some other handy pre-commit checks. We can add/remove some if you like.

Also fix the zizmor error found

## Test Plan

```console
❯ uv pip install pre-commit
...
❯ pre-commit run --all-files
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check for merge conflicts................................................Passed
check toml...............................................................Passed
check yaml...............................................................Passed
debug statements (python)................................................Passed
fix end of files.........................................................Passed
forbid submodules....................................(no files to check)Skipped
trim trailing whitespace.................................................Passed
Validate GitHub Workflows................................................Passed
zizmor...................................................................Passed
Validate pyproject.toml..................................................Passed
Check hooks apply to the repository......................................Passed
Check for useless excludes...............................................Passed
```

### CODE REVIEW

**Comment by graingert** [comment]
Probably we want ruff check/format also?

**Review by woodruffw — COMMENTED** [review]
Thanks, much appreciated!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,4 +1,11 @@
 repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.8.2
+    hooks:
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
```
Comment (graingert) [inline .pre-commit-config.yaml:6]: Is exit-non-zero-on-fix needed for pre-commit? I thought pre-commit failed if anything changed on disk?

Comment (hugovk) [reply .pre-commit-config.yaml]: Ah, right you are! Removed :)

**Review by woodruffw — APPROVED** [review]
LGTM, will merge tonight (can't merge on mobile because this updates a workflow file).

**Comment by graingert** [comment]
Hmm my mobile let's me merge it

**Comment by graingert** [comment]
How do I configure it now to allow that?

**Comment by woodruffw** [comment]
> How do I configure it now to allow that?

Are you using the app, or mobile website? If it's the app I think the workflow merge restriction happens because with OAuth logins, since it's technically an OAuth app/session and not a fully privileged user session.

**Pull request merged**